### PR TITLE
Add JS statements compat data

### PR DESCRIPTION
**Pull request by Elchi3**
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements and subpages.

### CODE REVIEW

File: javascript/statements.json [hunk 1]
```diff
@@ -0,0 +1,2214 @@
...
+              "version_added": true
+            },
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": true
```
Comment (teoli2003) [inline javascript/statements.json]: According to the table this is "6"

File: javascript/statements.json [hunk 2]
```diff
@@ -0,0 +1,2214 @@
...
+            },
+            "firefox_android": {
+              "version_added": true
+            },
+            "ie": {
+              "version_added": "6"
```
Comment (teoli2003) [inline javascript/statements.json]: According to the table this is 'true'

File: javascript/statements.json [hunk 3]
```diff
@@ -0,0 +1,2214 @@
...
+              "version_added": "55"
+            },
+            "edge": {
+              "version_added": true
+            },
+            "edge_mobile": {
+              "version_added": true
+            },
+            "firefox": {
+              "version_added": "52"
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/statements.json]: I think we shouldn't add this. Function expression closures are super non-standard. No one knows what they are or uses them and we will remove them soon hopefully.

File: javascript/statements.json [hunk 4]
```diff
@@ -0,0 +1,2214 @@
...
+            },
+            "firefox_android": {
+              "version_added": "52"
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/statements.json]: same

File: javascript/statements.json [hunk 5]
```diff
@@ -0,0 +1,2214 @@
...
+        },
+        "reassignment_fails": {
+          "__compat": {
+            "description": "Reassignment fails",
+            "support": {
+              "webview_android": {
+                "version_added": true
+              },
+              "chrome": {
+                "version_added": "20"
```
[the remark this answers is not in the document]
Comment (Elchi3) [reply javascript/statements.json]: Yes, this makes sense to me.

Also removed the "sloppy mode" entry. I think it is non-sense (I removed it elsewhere already).

File: javascript/statements.json (hunk 1 above)
Comment (Elchi3) [reply javascript/statements.json]: I thought this is mismatched with IE. No way only such a relatively late version of Firefox got try...catch.

File: javascript/statements.json (hunk 2 above)
Comment (Elchi3) [reply javascript/statements.json]: see above

File: javascript/statements.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (teoli2003) [reply javascript/statements.json]: Sounds fine then.

File: javascript/statements.json [hunk 6]
```diff
@@ -433,114 +443,6 @@
...
-              }
-            },
-            "status": {
-              "experimental": false,
-              "standard_track": true,
-              "deprecated": false
-            }
-          }
-        },
-        "allow_in_sloppy_mode": {
```
Comment (teoli2003) [inline javascript/statements.json]: @elchi3 Didn't you removed too much here? Shouldn't "allow_in_sloppy_mode" be kept?

Comment (Elchi3) [reply javascript/statements.json]: See the above comment.
> Also removed the "sloppy mode" entry. I think it is non-sense (I removed it elsewhere already).

Comment (teoli2003) [reply javascript/statements.json]: Oh missed it. Fine with it.

**Pull request merged**